database-mongo/index.js

```diff
@@ -13,22 +13,117 @@ db.once('open', function() {
 });
 
 var games = mongoose.Schema({
-  token: String,
-  players: String,
-  results: String  
+  token: {type: String, required: true},	
+  playerRoles: String,
+  results: String, 
+  playerIds: String,
 });
 
-var Game = mongoose.model('Game', games);
+Game = mongoose.model('Game', games);
 
-var selectGame = function(token, callback) {
-  Game.find({token});
+module.exports.selectGame = function(token, callback) {
+  Game.find({token}, (err, game) => {
+  	callback(game[0]); //game object with token, results, playerIdMapping, playerRolesMapping etc.
+  });
 };
 
-var createGame = function(token, players, results, callback) {
+module.exports.createGame = function(token, hostName, clientId, callback) {
+	Game.create({token}, (err) => {
+		if (err) {
+			callback(err);
+		} else {
+			var playerIds = {};
+			playerIds[hostName] = clientId;
+			playerIds = JSON.stringify(playerIds);
+			var results = [];
+			results = JSON.stringify(results);
+			Game.update({token}, {playerIds, results}, (err) => {
+				callback(err);
+			});
+		}
+	});
 };
 
-var updateResults = function (results, callback) {
+module.exports.addPlayer = function(token, playerName, clientId, callback) {
+	Game.find({token}, (err, game) => {
+		var playerIds = JSON.parse(game[0].playerIds);
+		playerIds[playerName] = clientId;
+		playerIds = JSON.stringify(playerIds);
+		Game.update({token}, {playerIds}, (err) => {
+			callback(err);
+		});
+	});
 };
 
+module.exports.removePlayer = function(token, playerName, callback) {
+	Game.find({token}, (err, game) => {
+		var playerIds = JSON.parse(game[0].playerIds);
+		delete playerIds[playerName];
+		playerIds = JSON.stringify(playerIds);
+		Game.update({token}, {playerIds}, (err) => {
+			callback(err);
+		});
+	});
+};
+
+module.exports.addRoles = function(token, playerRoles, callback) {
+	var playerRoles = JSON.stringify(playerRoles); 
+	Game.update({token}, {playerRoles}, (err) => {
+		callback(err);
+	});
+};
+
+module.exports.updateResults = function (token, roundResults, callback) {
+	Game.find({token}, (err, game) => {
+		console.log('got here')
+		var results = JSON.parse(game[0].results);
+		results.push(roundResults);
+		results = JSON.stringify(results);
+		Game.update({token}, {results}, (err) => {
+			callback(err);
+		})
+	});
+};
+
```

Comment:
From createGame at line 30 to here, callbacks are only called with err. Is this because WIP? If not, could you explain the design decision?